app/views/AnalyticalFramework/UserTable/AddUserModal/index.tsx

```diff
@@ -134,64 +202,56 @@ function AddUserModal(props: Props) {
             )
     ), [analysisFrameworkRolesResponse, isPrivateFramework]);
 
-    const {
-        pending: pendingAddAction,
-        trigger: triggerAddFrameworkMember,
-    } = useLazyRequest<unknown, ValueToSend>({
-        url: isDefined(userValue)
-            ? `server://framework-memberships/${userValue.id}/`
-            : 'server://framework-memberships/',
-        method: isDefined(userValue)
-            ? 'PATCH'
-            : 'POST',
-        body: (ctx) => ctx,
-        onSuccess: () => {
-            alert.show(
-                userValue?.id
-                    ? 'Successfully updated user permissions.'
-                    : 'Successfully added user to the analytical framework.',
-                {
-                    variant: 'success',
-                },
-            );
-            onTableReload();
-            onModalClose();
-        },
-        failureMessage: _ts('analyticalFramework.addUser', 'membershipPostFailed'),
-    });
-
     const handleSubmit = useCallback(
         () => {
             const submit = createSubmitHandler(
                 validate,
                 setError,
                 (val) => {
-                    const newVal = val.member ? { member: Number(val.member), ...val } : val;
-                    triggerAddFrameworkMember({
-                        ...newVal,
-                        framework: frameworkId,
-                    } as ValueToSend);
+                    if (isNotDefined(val.member)) {
+                        addEditFrameworkMembership({ variables: undefined });
```

Comment:
I don't understand why this case exists.
We either add a user or update it. It is defined by member part of the data.

Reply:
due to type issue we have to check it